synapse/lib/stormlib/aha.py

```diff
@@ -161,10 +258,210 @@ async def _del(self, svcname):
+                $lib.print(`Resolved {$cmdopts.svc} to an AHA Pool.\n`)
+                $lib.print(`The pool currently has {$lib.len($services)} members.`)
+
+                $lib.print(`AHA Pool:   {$svc.name}`)
+                for ($_svcname, $_svcinfo) in $services {
+                    $lib.print(`Member:     {$_svcname}`)
+                }
+            } else {
+                $lib.print(`Resolved {$cmdopts.svc} to an AHA Service.\n`)
+                $svcinfo = $svc.svcinfo
+                $leader = $svcinfo.leader
+                if ($leader = (null)) {
+                    $leader = 'Service did not register itself with a leader name.'
+                }
+                $lib.print(`Name:       {$svc.name}`)
+                $lib.print(`Online:     {$_getBoolKey($svcinfo, online, false)}`)
+                $lib.print(`Ready:      {$_getBoolKey($svcinfo, ready, null)}`)
+                $lib.print(`Run iden:   {$svcinfo.run}`)
+                $lib.print(`Cell iden:  {$svcinfo.iden}`)
+                $lib.print(`Leader:     {$leader}`)
+
+                if $cmdopts.nexus {
+                    if $svcinfo.online {
+                        $nexusOffset = $_getNexus($svc.name)
+                    } else {
+                        $nexusOffset = 'Service is not online. Will not attempt to retrieve its nexus offset.'
+                    }
+                    $lib.print(`Nexus:      {$nexusOffset}`)
+                }
+
+                $lib.print('Connection information:')
+                $urlinfo = $svcinfo.urlinfo
+                $keys = $lib.dict.keys($urlinfo)
+                $keys.sort()
+                for $k in $keys {
+                    $dk = `{$k}:`
+                    $dk = $dk.ljust(12)
+                    $lib.print(`    {$dk}{$urlinfo.$k}`)
+                }
+            }
+        }
+        '''
+    },
+    {
+        'name': 'aha.svc.list',
+        'descr': '''List AHA services.
+
+If the --nexus argument is given, the Cortex will attempt to connect to each service and report the Nexus offset of the service.
+
+The ready column indicates that a service has entered into the realtime change window for synchronizing changes from its leader.''',
+        'cmdargs': (
+            ('--nexus', {'help': 'Try to connect to online services and report their nexus offset.',
+                         'default': False, 'action': 'store_true'}),
+        ),
+        'storm': '''
+        function _getNexus(svcname) {
+            $_url = `aha://{$svcname}/`
+            try {
+                $_prox = $lib.telepath.open($_url)
+                $_info = $_prox.getCellInfo()
+                return ( $_info.cell.nexsindx )
+            } catch * as _err {
+                $_emsg = $_err.mesg
+                if ($_emsg = $lib.null ) {
+                    $_emsg = `{$_err}`
+                }
+                return ( $_emsg )
+            }
+        }
+
+        function _getBoolKey(dict, key, default) {
```

Comment:
This feels a bit hacky. Is this necessary to prevent python True/False strings?

Reply:
trying to prevent potentially unclear `$lib.null` strings from filtering into output.